### PR TITLE
Skip file if flake8 fails to run

### LINKED ISSUE #133: If flake8 fails, then yesqa removes codes
If the `flake8` command fails (say, due to some bug in a plugin), then `yesqa` will remove the noqa codes.

Reproducible example: 
- install make `yesqa` pre-commit hook with additional dependencies: `flake8==5.0.4, flake8-bugbear==21.3.2`
- on the file `t.py`
```python
def test_hash_collisions():
    hashes = [
        "Ingrid-9Z9fKIZmkO7i7Cn51Li34pJm44fgX6DYGBNj3VPlOH50m7HnBlPxfIwFMrcNJNMP6PSgLmwWnInciMWrCSAlLEvt7JkJl4IxiMrVbXSa8ZQoVaq5xoQPjltuJEfwdNlO6jo8qRRHvD8sBEBMQASrRa6TsdaPTPCBo3nwIBpE7YzzmyH0vMBhjQZLx1aCT7faSEx7PgFxQhHdKFWROcysamgy9iVj8DO2Fmwg1NNl93rIAqC3mdqfrCxrzfvIY8aJdzin2cHVzy3QUJxZgHvtUtOLxoqnUHsYbNTeq0xcLXpTZEZCxD4PGubIuCNf32c33M7HFsnjWSEjE2yVdWKhmSVodyF8hFYVmhYnMCztQnJrt3O8ZvVRXd5IKwlLexiSp4h888w7SzAIcKgc3g5XQJf6MlSMftDXm9lIsE1mJNiJEv6uY6pgvC3fUPhatlR5JPpVAHNSbSEE73MBzJrhCAbOLXQumyOXigZuPoME7QgJcBalliQol7YZ9",  # noqa: E501
        "Tim-b9MddTxOWW2AT1Py6vtVbZwGAmYCjbp89p8mxsiFoVX4FyDOF3wFiAkyQTUgwg9sVqVYOZo09Dh1AzhFHbgij52ylF0SEwgzjzHH8TGY8Lypart4p4onnDoDvVMBa0kdthVGKl6K0BDVGzyOXPXKpmnMF1H6rJzqHJ0HywfwS4XYpVwlAkoeNsiicHkJUFdUAhG229INzvIAiJuAHeJDUoyO4DCBqtoZ5TDend6TK7Y914yHlfH3g1WZu5LksKv68VQHJriWFYusW5e6ZZ6dKaMjTwEGuRgdT66iU5nqWTHRH8WSzpXoCFwGcTOwyuqPSe0fTe21DVtJn1FKj9F9nEnR9xOvJUO7E0piCIF4Ad9yAIDY4DBimpsTfKXCu1vdHpKYerzbndfuFe5AhfMduLYZJi5iAw8qKSwR5h86ttXV0Mc0QmXz8dsRvDgxjXSmupPxBggdlqUlC828hXiTPD7am0yETBV0F3bEtvPiNJfremszcV8NcqAoARMe",  # noqa: E501
    ]
```
run `pre-commit run yesqa --files t.py`

The hook exits with `1`, and the `# noqa: E501` comments have been removed

Noticed in https://github.com/pandas-dev/pandas/pull/47913

### PR DESCRIPTION
**Pull request by MarcoGorelli**
closes #133

This assumes `flake8` doesn't write to stderr, which I don't think it does

### CODE REVIEW

File: yesqa.py [hunk 1]
```diff
@@ -20,17 +20,23 @@
...
-        filename,
-    )
-    out, _ = subprocess.Popen(cmd, stdout=subprocess.PIPE).communicate()
+    cmd = (*CMD, filename)
+    out, err = subprocess.Popen(
+        cmd,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+    ).communicate()
+    if err:
```
Comment (asottile) [inline yesqa.py:38]: stderr alone doesn't mean flake8 "failed" -- for example warnings

**Comment by asottile** [comment]
this all seems honestly too fragile and not worth fixing, sorry :/

**Comment by asottile** [comment]
`--exit-zero` + nonzero exit code might be better? but I kinda want to kill that option out of `flake8`

**Comment by MarcoGorelli** [comment]
Ok no worries, I was just very confused earlier when I saw all those noqa E501 being removed

**Pull request closed without merging**